subprojects/core/src/main/java/org/gradle/api/internal/FeaturePreviews.java

```diff
@@ -17,18 +17,39 @@
 
 import org.gradle.StartParameter;
 
+import java.util.EnumSet;
+
 public class FeaturePreviews {
-    private StartParameter startParameter;
+    private enum Feature {
+        ADVANCED_POM_SUPPORT,
```

Comment:
We agreed that this was a bad name, and that `IMPROVED_POM_SUPPORT` was better (even if it's not ideal). We should make this change now.

Reply:
Consider it done